reactExercise/solutions/zihe/ex5/src/UseEffect.js

```diff
@@ -0,0 +1,54 @@
+import React, { useEffect, useState } from 'react'
+
+const UseEffect = () => {
+    const [y1, setY1] = useState(false)
+    const [y2, setY2] = useState(false)
+    const [y3, setY3] = useState(false)
+    const [isloading, setIsLoading] = useState(true)
+
+    useEffect(() => {
+        setTimeout(() => { setIsLoading(false) }, 3000)
```

Comment:
如果我们添加一个clean up function 可以实现 componentWillUnmount 的效果  你试试
```suggestion
        setTimeout(() => { setIsLoading(false) }, 3000)
        return ()=> alert('sure?')
```